web/static/js/components/surveys/SurveyShow.jsx

```diff
@@ -51,6 +51,30 @@ class SurveyShow extends Component {
       return <p>Loading...</p>
     }
 
+    let primaryMode = <div className='mode'>
+      <span className='type'>Primary Mode</span>
+      <div>
+        <i className='material-icons'>{survey.mode[0] == 'sms' ? 'sms' : 'phone'}</i>
+        <span className='mode-label name'>{survey.mode[0] == 'sms' ? 'SMS' : 'IVR'}</span>
+      </div>
+    </div>
+
+    let fallbackMode = null
```

Comment:
Extract right handside of `let fallbackMode` to a function that receives `survey.mode`.